ocs_ci/ocs/pgsql.py

```diff
@@ -258,6 +283,75 @@ def validate_pgbench_run(self, pgbench_pods):
             all_pgbench_pods_output.append(pg_output)
         return all_pgbench_pods_output
 
+    def get_nodes(self, pod_name, all_nodes=False):
+        """
+        Get nodes that contain a specific pod
+
+        Args:
+            pod_name (str): pod name
+
+            all_nodes (bool):
+                True-return all the nodes that contain postgres pod
+                False-return 1 node that contain postgres pod
+
+        Returns:
+            list: Cluster node OCP objects
+
+        """
+        if pod_name == 'osd':
+            pods_obj = self.pod_obj.get(selector=constants.OSD_APP_LABEL, all_namespaces=True)
+        elif pod_name == 'postgres':
+            pods_obj = self.pod_obj.get(selector=constants.PGSQL_APP_LABEL, all_namespaces=True)
+
+        log.info(f"Create a list of nodes (without duplicate nodes in the list)")
+        nodes_set = set()
+        for pod in pods_obj['items']:
+            log.info(f"pod {pod['metadata']['name']} located on node {pod['spec']['nodeName']}")
+            nodes_set.add(pod['spec']['nodeName'])
+
+        if all_nodes:
+            node_list = get_node_objs(list(nodes_set))
+            return node_list
+        # Selects one Node (random)
+        node_obj = get_node_objs(list(nodes_set)[random.randint(0, len(nodes_set) - 1)])
+        log.info(f"Selects one Node (random) - {node_obj[0].name}")
+        return node_obj
+
+    def respin_pod(self, pod_name=''):
+        """
+        Respin a pod
+
+        Args:
+            pod_name (str)
+
+        Returns:
+            str: pod status
+
+        """
+        log.info(f"Respin pod {pod_name}")
+        if pod_name == 'postgers':
+            self.respin_app_pod()
+            return True
+        disruption = disruption_helpers.Disruptions()
+        disruption.set_resource(resource=f'{pod_name}')
+        disruption.delete_resource()
+
+    def respin_app_pod(self):
```

Comment:
Instead of picking a random app pod, can we take pod_obj as an arg/input and delete that specific pod.
WDYT?